core/proto/keytransparency_v1_grpc/keytransparency_v1_admin_grpc.proto

```diff
@@ -87,3 +74,69 @@ service KeyTransparencyAdminService {
     };
   }
 }
+
+// The AccountRecoveryService API holds a set of account recovery keys.  
+// Access to these APIS are logged and restricted to properly authenticated
+// and authorized callers.  Only accounts that have put one of the recovery
+// keys in their set of authorized_keys can be administered.  Third parties can
+// implement this API in order to provide trust agility with regard to account
+// reset.
+// 
+// Key Rotation: 
+// The set of account recovery keys may rotate over time with new keys being
+// added every X months, X * N months before they become the currently active signing
+// key, and another Y months before they are deactivated from the list of valid
+// verifying keys. In order to maintain the ability of this service to
+// administer accounts, they must refresh their set of authorized public keys,
+// ideally every X months.
+//
+// Resources:
+// - recovery/keysets
+//   Each keyset is restricted to a specific set of authenticated callers.
+service AccountRecoveryService {
+
+  // More than one keyset per endpoint? yes.
+  // Keyset IDs
+  // 1 keyset per domain? 
+  // 1 Keyset per domain + app
+  // 1 Keyset per authorization policy.  Could be restricted to domain + app caller.
+
+  // GetKeySet returns the requested keyset.
+  rpc GetKeySet (keytransparency.v1.proto.GetKeySetRequest) returns (keymaster.proto.KeySet) {
+    option (google.api.http) = {
+      get: "/v1/recovery/keysets/{keyset_id}"
+    };
+  }
+
+  // ListKeySets lists the available keysets this service has. 
+  rpc ListKeySets (keytransparency.v1.proto.ListKeySetsRequest) returns (keytransparency.v1.proto.ListKeySetsResponse) {
+    option (google.api.http) = {
+      get: "/v1/recovery/keysets"
+    };
+  }
+
+  // CreateAccount creates a new account and initializes it with data and authorized_keys.
+  // If the account already exists, this operation will fail.
+  rpc CreateAccount(keytransparency.v1.proto.CreateAccountRequest) returns (keytransparency.v1.proto.Account) {
+    option (google.api.http) = {
+      post: "/v1/recovery/keysets/{keyset_id}/domains/{account.domain_id}/apps/{account.app_id}/users/{account.user_id}"
+      body: "account"
+    };
+  }
+
+  // SetAccountData sets the public key for an account. 
+  rpc UpdateAccount(keytransparency.v1.proto.UpdateAccountRequest) returns (keytransparency.v1.proto.Account) {
+    option (google.api.http) = {
+      patch: "/v1/recovery/keysets/{keyset_id}/domains/{account.domain_id}/apps/{account.app_id}/users/{account.user_id}"
+      body: "account"
+    };
+  }
+
+  // BatchCreateAccount creates multiple accounts at once.
+  rpc BatchCreateAccount(keytransparency.v1.proto.BatchCreateAccountRequest) returns (keytransparency.v1.proto.BatchCreateAccountResponse) {
+    option (google.api.http) = {
+      post: "/v1/recovery/keysets/{keyset_id}:batch"
```

Comment:
Should domain_id be part of the url? How should that map onto repeated accounts?